presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/json/JsonFileBasedFunctionNamespaceManager.java

```diff
@@ -0,0 +1,209 @@
+import com.facebook.presto.spi.function.AlterRoutineCharacteristics;
+import com.facebook.presto.spi.function.FunctionMetadata;
+import com.facebook.presto.spi.function.Parameter;
+import com.facebook.presto.spi.function.ScalarFunctionImplementation;
+import com.facebook.presto.spi.function.SqlFunctionHandle;
+import com.facebook.presto.spi.function.SqlFunctionId;
+import com.facebook.presto.spi.function.SqlInvokedFunction;
+import com.google.common.collect.ImmutableList;
+
+import javax.inject.Inject;
+
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.facebook.presto.common.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.plugin.base.JsonUtils.parseJson;
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_USER_ERROR;
+import static com.facebook.presto.spi.StandardErrorCode.NOT_SUPPORTED;
+import static com.facebook.presto.spi.function.FunctionVersion.notVersioned;
+import static com.facebook.presto.spi.function.RoutineCharacteristics.Language.CPP;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.MoreCollectors.onlyElement;
+import static java.lang.Long.parseLong;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class JsonFileBasedFunctionNamespaceManager
+        extends AbstractSqlInvokedFunctionNamespaceManager
+{
+    private final Map<SqlFunctionId, SqlInvokedFunction> latestFunctions = new ConcurrentHashMap<>();
+    private final Map<QualifiedObjectName, UserDefinedType> userDefinedTypes = new ConcurrentHashMap<>();
+    private final JsonFileBasedFunctionNamespaceManagerConfig managerConfig;
+
+    @Inject
+    public JsonFileBasedFunctionNamespaceManager(
+            @ServingCatalog String catalogName,
+            SqlFunctionExecutors sqlFunctionExecutors,
+            SqlInvokedFunctionNamespaceManagerConfig config,
+            JsonFileBasedFunctionNamespaceManagerConfig managerConfig)
+    {
+        super(catalogName, sqlFunctionExecutors, config);
+        this.managerConfig = requireNonNull(managerConfig, "managerConfig is null");
+        bootStrapNameSpaceFromFile();
+    }
+
+    private static SqlInvokedFunction copyFunction(SqlInvokedFunction function)
+    {
+        return new SqlInvokedFunction(
+                function.getSignature().getName(),
+                function.getParameters(),
+                function.getSignature().getReturnType(),
+                function.getDescription(),
+                function.getRoutineCharacteristics(),
+                function.getBody(),
+                function.getVersion());
+    }
+
+    private void bootStrapNameSpaceFromFile()
```

Comment:
bootstrap and namespace are usually each considered one word (so the "s"s shouldn't be uppercase)